utils/prepare-contact-submit-data.ts

```diff
@@ -1,122 +1,112 @@
-import { ContactFormData, EmailAddressFormData, PhoneNumberFormData, StreetAddressFormData } from "../interfaces/redtail-form.interface";
+import {
+  RedtailContactUpdate,
+  ContactRecordUpdate,
+  AddressUpdate,
+  EmailUpdate,
+  PhoneUpdate,
+  UrlUpdate,
+} from "../interfaces/redtail-contact-update.interface";
+import {
+  RedtailContactRec,
+  AddressRec,
+  EmailRec,
+  PhoneRec,
+  UrlRec,
+} from "../interfaces/redtail-contact.interface";
+import { ContactFormData } from "../interfaces/redtail-form.interface";
 import { toRedtailDatestring } from "./redtail-datestrings";
 
 export const prepareContactSubmitData = (
   formData: ContactFormData,
-  fields: RedtailContactMasterRec
+  contactRec: RedtailContactRec
 ) => {
-
-  const contactRecord: ContactFieldsUpdate = {
-    AnniversaryDate: fields.ContactRecord.AnniversaryDate,
-    CategoryID: formData.categoryID,
-    ClientID: fields.ContactRecord.ClientID,
-    ClientSinceDate: fields.ContactRecord.ClientSinceDate,
-    CompanyID: fields.ContactRecord.CompanyID,
-    DateOfBirth: toRedtailDatestring(formData.dateOfBirth),
-    Designation: fields.ContactRecord.Designation,
-    Family: fields.ContactRecord.Family,
-    FamilyHeadID: fields.ContactRecord.FamilyHeadID,
-    Familyname: fields.ContactRecord.Familyname,
-    Firstname: formData.firstName,
-    Gender: formData.gender,
-    JobTitle: fields.ContactRecord.JobTitle,
-    Lastname: formData.lastName,
-    MaritalStatus: fields.ContactRecord.MaritalStatus,
-    Middlename: formData.middleName,
-    Name: fields.ContactRecord.Name,
-    Nickname: formData.nickname,
-    ReferredBy: fields.ContactRecord.ReferredBy,
-    Salutation: formData.salutation,
-    ServicingAdvisorID: formData.servicingAdvisorID,
-    SourceID: formData.sourceID,
-    StatusID: formData.statusID,
-    Suffix: fields.ContactRecord.Suffix,
-    TaxID: formData.taxID,
-    TypeID: fields.ContactRecord.ClientType,
-    WritingAdvisorID: formData.writingAdvisorID,
+  const contactRecord: ContactRecordUpdate = {
+    id: contactRec.id,
+    salutation_id: formData.salutationID,
+    first_name: formData.firstName,
+    middle_name: formData.middleName,
+    last_name: formData.lastName,
+    nickname: formData.nickname,
+    gender_id: formData.genderID,
+    dob: formData.dateOfBirth,
+    category_id: formData.categoryID,
+    status_id: formData.statusID,
+    source_id: formData.sourceID,
+    tax_id: formData.taxID,
+    servicing_advisor_id: formData.servicingAdvisorID,
+    writing_advisor_id: formData.writingAdvisorID,
   };
 
-  const internet: InternetUpdate[] = formData.emailAddresses.map((item) => {
-
-    const fromSource:
-      | InternetRec
-      | undefined = fields.Internet.find(
-      (i) => i.RecID === item.recID
+  const addresses: AddressUpdate[] = formData.addresses.map((item) => {
+    const fromSource: AddressRec | undefined = contactRec.addresses.find(
```

Comment:
It doesn't look like we're checking if the data from each item is fromSource or not anymore. Can we check if this is still needed?